pkg/ccl/logictestccl/tests/local-legacy-schema-changer/local-legacy-schema-changer_test: TestCCLLogic_select_for_update_read_committed failed

pkg/ccl/logictestccl/tests/local-legacy-schema-changer/local-legacy-schema-changer_test.TestCCLLogic_select_for_update_read_committed [failed](https://tanzanite.cluster.engflow.com/invocations/default/1e1ea698-cf0d-45aa-a711-1bcc818df954?testReportRun=5&testReportShard=22&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbC1sZWdhY3ktc2NoZW1hLWNoYW5nZXI6bG9jYWwtbGVnYWN5LXNjaGVtYS1jaGFuZ2VyX3Rlc3Q=) on master @ [dbc62db07a1f149d305cff463b350588086f1b08](https://github.com/cockroachdb/cockroach/commits/dbc62db07a1f149d305cff463b350588086f1b08):


```
=== RUN   TestCCLLogic_select_for_update_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_select_for_update_read_committed2093978617
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    logic.go:4317: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_select_for_update_read_committed2093978617
--- FAIL: TestCCLLogic_select_for_update_read_committed (0.88s)
=== RUN   TestCCLLogic_select_for_update_read_committed/skip_locked
    logic.go:2967: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-legacy-schema-changer/local-legacy-schema-changer_test_/local-legacy-schema-changer_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:616: SELECT * FROM xyz WHERE z = 100 ORDER BY x FOR UPDATE SKIP LOCKED
        expected:
            2  10  100
            
        but found (query options: "") :
            
    logic.go:2205: 
         /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-legacy-schema-changer/local-legacy-schema-changer_test_/local-legacy-schema-changer_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed:621: too many errors encountered, skipping the rest of the input
[07:55:04] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-legacy-schema-changer/local-legacy-schema-changer_test_/local-legacy-schema-changer_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed with config local-legacy-schema-changer: 109 tests, 2 failures
--- total: 109 tests, 2 failures
    --- FAIL: TestCCLLogic_select_for_update_read_committed/skip_locked (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=22</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_select_for_update_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40979

--- a/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed
+++ b/pkg/ccl/logictestccl/testdata/logic_test/select_for_update_read_committed
@@ -571,7 +571,7 @@ statement ok
 GRANT ALL ON xyz TO testuser
 
 statement ok
-INSERT INTO xyz VALUES (1, 10, 100), (2, 10, 100)
+INSERT INTO xyz VALUES (1, 10, 100), (2, 10, 200), (3, 30, 100)
 
 user testuser
 
@@ -601,7 +601,7 @@ BEGIN ISOLATION LEVEL READ COMMITTED;
 SELECT * FROM xyz@xyz_y_idx WHERE y = 10 ORDER BY y, x LIMIT 1 FOR UPDATE SKIP LOCKED;
 COMMIT;
 ----
-2  10  100
+2  10  200
 
 # The unlocked reads in SKIP LOCKED should not block on locks, either, even
 # under serializable isolation.
@@ -615,7 +615,7 @@ SET enable_durable_locking_for_serializable = true
 query III
 SELECT * FROM xyz WHERE z = 100 ORDER BY x FOR UPDATE SKIP LOCKED
 ----
-2  10  100
+3  30  100
 
 user testuser
 

